### ISSUE TITLE
[joss] graphviz_draw() bad output

### ISSUE THREAD

**Issue by szhorvat** [Qiskit/retworkx#526]
`graphviz_draw()` always places an additional, non-existent node in the drawing. Example:

```
graphviz_draw(rx.generators.cycle_graph(3))
```

<img width="218" alt="image" src="https://user-images.githubusercontent.com/1212871/144755918-dd93fbd1-2cbe-4358-be58-ed548d4ee0bb.png">

I have retworkx 0.10.2 and pydot 1.4.2.

**Comment by mtreinish** [comment]
Hmm, I'm able to reproduce this locally with 0.10.2 from pypi. But when I run that code with 0.11.0 locally I get:

![cycle](https://user-images.githubusercontent.com/2447371/144760254-1267cdad-0a5d-4ed1-97e9-40091e018ae0.png)

Interestingly looking at the dot generated by `rx.generators.cycle_graph(3).to_dot()` which is what `graphviz_draw()` uses internally yields the same on both main and 0.10.2:

```dot
graph {
0 ;
1 ;
2 ;
0 -- 1 ;
1 -- 2 ;
2 -- 0 ;
}

```

which is the correct dot output for the method with `cycle_graph(3)`. So I'm a bit at a loss for what's going on since the drawer code hasn't changed at all on main since the 0.9.0 release and the `to_dot()` method hasn't had any changes on `main` since before the 0.10.0 release. My only thought is something with pydot is parsing the dot string, but both my 0.11.0/main run and the 0.10.2 version are on the same pydot version, 1.4.2.

**Comment by mtreinish** [comment]
Hmm, playing with this a little more I think it is a difference in the pydot parsing, although I'm still not sure why it's happening. When I run `pydot.graph_from_dot_data()` on the dot string from `rx.generators.cycle_graph(3).to_dot()` on a fresh `pydot` install the output `pydot` is parsing is:
```dot
graph G {
0;
1;
2;
0 -- 1;
1 -- 2;
2 -- 0;
"\n";
}
```
That extra `"\n";` line that pydot is adding is the source of the blank node. I don't know why it's doing that though, but it has been reported upstream already: https://github.com/pydot/pydot/issues/280 given that it's only been noticed recently (both by us and the reported for the upstream issue) my guess is it's a compatibility issue with `pyparsing` which has had recent releases in the past month (including a new major version in late October) which is the package pydot uses for parsing dot files.

**Comment by mtreinish** [comment]
Yep, I just confirmed that installing pyparsing <= 3.0.1 fixes this issue (3.0.2 fails to parse the dot string) and pyparsing >=3.0.3 adds the extra `"\n"`.

**Comment by szhorvat** [comment]
The `.to_dot()` output is indeed correct. So the difference was the pyparsing version in your virtualenvs.
